cypress-tests/cypress/e2e/PaymentUtils/Noon.js

```diff
@@ -0,0 +1,693 @@
+    card_holder_name: null,
+    payment_checks: {
+      avs_response: {
+        code: "Y",
+        codeRaw: "Y",
+      },
+      card_verification: null,
+    },
+    authentication_data: null,
+  },
+  billing: null,
+};
+
+const payment_method_data_3ds = {
+  card: {
+    last4: "1091",
+    card_type: "CREDIT",
+    card_network: "Visa",
+    card_issuer: "INTL HDQTRS-CENTER OWNED",
+    card_issuing_country: "UNITEDSTATES",
+    card_isin: "400000",
+    card_extended_bin: null,
+    card_exp_month: "01",
+    card_exp_year: "25",
+    card_holder_name: null,
+    payment_checks: null,
+    authentication_data: null,
+  },
+  billing: null,
+};
+
+export const connectorDetails = {
+  card_pm: {
+    PaymentIntent: {
+      Request: {
+        currency: "AED",
+        customer_acceptance: null,
+        setup_future_usage: "on_session",
+        connector_metadata: connectorMetadata,
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "requires_payment_method",
+          setup_future_usage: "on_session",
+        },
+      },
+    },
+    PaymentIntentOffSession: {
+      Request: {
+        currency: "AED",
+        amount: 6500,
+        authentication_type: "no_three_ds",
+        customer_acceptance: null,
+        setup_future_usage: "off_session",
+        connector_metadata: connectorMetadata,
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "requires_payment_method",
+          setup_future_usage: "off_session",
+        },
+      },
+    },
+    "3DSManualCapture": {
+      Request: {
+        payment_method: "card",
+        currency: "AED",
+        payment_method_data: {
+          card: successfulThreeDSTestCardDetails,
+        },
+        customer_acceptance: null,
+        setup_future_usage: "on_session",
+        connector_metadata: connectorMetadata,
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "requires_customer_action",
+          setup_future_usage: "on_session",
+          payment_method_data: payment_method_data_3ds,
+        },
+      },
+    },
+    "3DSAutoCapture": {
+      Request: {
+        payment_method: "card",
+        payment_method_data: {
+          card: successfulThreeDSTestCardDetails,
+        },
+        currency: "AED",
+        customer_acceptance: null,
+        connector_metadata: connectorMetadata,
+        setup_future_usage: "on_session",
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "requires_customer_action",
+          setup_future_usage: "on_session",
+          payment_method_data: payment_method_data_3ds,
+        },
+      },
+    },
+    No3DSManualCapture: {
+      Request: {
+        payment_method: "card",
+        payment_method_data: {
+          card: successfulNo3DSCardDetails,
+        },
+        connector_metadata: connectorMetadata,
+        customer_acceptance: null,
+        currency: "AED",
```

Comment:
Please do not pass currency in confirm calls.
I do not think we need it in refunds either.

Reply:
We need to pass currency in some of the scenarios where we use create + confirm

Reply:
Create + confirm, okay. But in every other case.

Reply:
Even Refund Payments use create + confirm